src/BreakpointViewer.cpp

```diff
@@ -0,0 +1,621 @@
+#include <QPainter>
+#include <QPaintEvent>
+#include <QMessageBox>
+#include <QToolTip>
+#include <QComboBox>
+#include <QDebug>
+#include <cassert>
+#include "BreakpointViewer.h"
+#include "ui_BreakpointViewer.h"
+#include "Convert.h"
+#include "CommClient.h"
+#include "OpenMSXConnection.h"
+
+enum TableColumns {
+	ENABLED = 0,
+	WP_TYPE = 1,
+	LOCATION = 2,
+	BP_ADDRESS = 2,
+	WP_REGION = 2,
+	CONDITION = 3,
+	SLOT = 4,
+	SEGMENT = 5,
+};
+
+struct ItemPos {
+	BreakpointViewer::Type type;
+	int row;
+	bool operator==(const ItemPos &ip) const
+	{
+		return (ip.type == type) && (ip.row == row);
+	}
+};
+
+QList<ItemPos> items;
```

Comment:
I don't like this, a global variable. Could you try to find a better place for this? I haven't yet studied your code in enough detail to be able to make a good suggestion.

Most of my review comments are about style or are mere suggestions. But this one is a blocker to merge your PR.

Reply:
This should be 
```
static QList<ItemPos> items;
```
Changing it.

Reply:
Better, but that's still a global variable (just with no visibility from other translation units).
I think this should become a member variable of the BreakpointViewer class.

ATM this distinction may seem moot. But who knows, maybe in the future we may want to support saving/loading a set of breakpoints. Or attach to multiple openMSX sessions at once (each with it own set of breakpoints). I know maybe far fetched examples, but future changes like these are much easier if the data is, already now, more organized. So NO global variables.

Reply:
OK. I was coding under the assumption that its better to leave implementation details out of the header in favour of a leaner presentation, but that reasoning makes sense.